src/app/analytics/analytics.factory.js

```diff
@@ -334,11 +335,13 @@ function analyticsFactory($window, $timeout, sessionService) {
         // Build cart data layer
         this.setDonorDetails(donorDetails);
         this.buildProductVar(cartData);
-
-        var aaProducts = $window.s && $window.s.setProducts('checkout');
-
-        // Store data for use on following page load
-        localStorage.setItem('aaProducts', aaProducts);
+      }catch(e){
+        // Error caught in analyticsFactory.purchase
+      }
+    },
+    setPurchaseNumber: function(purchaseNumber) {
+      try {
+        $window.digitalData.purchaseNumber = purchaseNumber;
       }catch(e){
         // Error caught in analyticsFactory.purchase
```

Comment:
This comment is now wrong... I kept the comments cuz eslint didn't like empty catch statements... Doesn't matter much...